### PR TITLE
add patchview to nwb tools gallery

### PR DESCRIPTION
**Pull request by ZeitgeberH**
Hi, this is a PR for PatchView to be included in the overview gallery.  Thanks!

### CODE REVIEW

File: docs/source/tools/patchview/patchview.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+.. image:: patchview.png
+    :class: align-center
+    :width: 600
+
+-----
+
+NWB (NeurodataWithoutBorders) APIs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+PatchView read Heka or Axon file format for patch-clamp data, and use `pynwb <https://github.com/NeurodataWithoutBorders/pynwb>`_ to exports to NWB files.
```
Comment (oruebel) [inline docs/source/tools/patchview/patchview.rst:23]:
```suggestion
PatchView reads Heka or Axon format files for patch-clamp data, and uses `PyNWB <https://github.com/NeurodataWithoutBorders/pynwb>`_ to export to NWB files.
```

File: docs/source/tools/patchview/patchview.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Code snippet shows how to convert data from original Heka .dat file into an extended `NWBFile` object, query the content and generate a visualization.
```
Comment (oruebel) [inline docs/source/tools/patchview/patchview.rst:25]:
```suggestion
The code snippet below shows how to convert data from an original Heka .dat file to an extended `NWBFile` object, query the content, and generate a visualization.
```

**Comment by oruebel** [comment]
Thanks for contributing this tool! The pages look good to me. I just added a few very minor suggestion.

To preview how the pages you added will appear on the website you can select the "Show all checks" link on this page and then select the "Details" link next to the ``docs/readthedocs:nwb-overview`` test. E.g., here are the current links so you can check that the pages look good:

https://nwb-overview--70.org.readthedocs.build/en/70/tools/patchview/patchview.html
https://nwb-overview--70.org.readthedocs.build/en/70/tools/tools_home.html#intracellular-electrical-physiology-tools

**Comment by oruebel** [comment]
Does PatchView also support reading from NWB files? In the text you mention that PatchView supports reading from Heka or Axon and then export to NWB. Can PatchView then also read from the NWB files again after export?

File: docs/source/tools/patchview/patchview.rst (hunk 2 above)
Comment (ZeitgeberH) [reply docs/source/tools/patchview/patchview.rst:25]: Looks good. Thanks for the edit!

**Review by oruebel — APPROVED** [review]

**Comment by ZeitgeberH** [comment]
> Does PatchView also support reading from NWB files? In the text you mention that PatchView supports reading from Heka or Axon and then export to NWB. Can PatchView then also read from the NWB files again after export?

Yes, internally,  PatchView use `NWBHDF5IO` for `pynwb` for writing and reading NWB files. Only the `pynwb` object, but not the  extra functions from the extended PatchView class would  be saved in this case.  Alternatively,   one can save the file into [dill.pickles](https://github.com/uqfoundation/dill) to preserve the extended functionality, but reading it back need an import of the original PatchView class first.

**Comment by oruebel** [comment]
> but not the extra functions from the extended PatchView class would be saved in this case

What sort of extended information does PatchView save? Depending on the kind of metadata you may be able to store the data as custom columns, e.g,. in the IntracelluarRecordingsTable. If the data you need to store is not supported by the core NWB data model, then you could create an extension for NWB to be able to store all the information from PatchView in NWB. Here a tutorial on how to create extensions: https://nwb-overview.readthedocs.io/en/latest/extensions_tutorial/extensions_tutorial_home.html

**Comment by ZeitgeberH** [comment]
> > but not the extra functions from the extended PatchView class would be saved in this case
> 
> What sort of extended information does PatchView save? Depending on the kind of metadata you may be able to store the data as custom columns, e.g,. in the IntracelluarRecordingsTable. If the data you need to store is not supported by the core NWB data model, then you could create an extension for NWB to be able to store all the information from PatchView in NWB. Here a tutorial on how to create extensions: https://nwb-overview.readthedocs.io/en/latest/extensions_tutorial/extensions_tutorial_home.html

Thanks for the suggestion.
For a particular neuron, I may have its intracelluar responses (NWB sweep table), morphology (a  neurolucida .ASC file) and transcriptome profile (a sparse matrix).  I plan to use PatchView's interface to integrate data from these three modalities, then save it as NWB file.
The NWB extension looks like exactly what I want.  Will look into it.

**Comment by oruebel** [comment]
> morphology (a neurolucida .ASC file)

Having support for morphology would be great to see in NWB. I think the SONATA extension could potentially be relevant as it also needs to describe compartments etc. of a neuron: https://github.com/catalystneuro/ndx-simulation-output

> transcriptome profile (a sparse matrix)

Here the sparse matrix type in the HDMF common schema (which NWB includes) could be relevant for storing sparse matrices: https://hdmf-common-schema.readthedocs.io/en/latest/format.html?highlight=Sparse#csrmatrix

**Pull request merged**